packages/app-store/pipedrive-crm/lib/CalendarService.ts

```diff
@@ -0,0 +1,313 @@
+import { getLocation } from "@calcom/lib/CalEventParser";
+import logger from "@calcom/lib/logger";
+import type {
+  Calendar,
+  CalendarEvent,
+  EventBusyDate,
+  IntegrationCalendar,
+  NewCalendarEventType,
+  Person,
+} from "@calcom/types/Calendar";
+import type { CredentialPayload } from "@calcom/types/Credential";
+
+import appConfig from "../config.json";
+
+type ContactSearchResult = {
+  status: string;
+  results: Array<{
+    id: string;
+    email: string;
+    firstName: string;
+    lastName: string;
+    name: string;
+  }>;
+};
+
+type ContactCreateResult = {
+  status: string;
+  result: {
+    id: string;
+    email: string;
+    firstName: string;
+    lastName: string;
+    name: string;
+  };
+};
+
+export default class PipedriveCalendarService implements Calendar {
+  private log: typeof logger;
+  private tenantId: string;
+  private revertApiKey: string;
+  private revertApiUrl: string;
+  constructor(credential: CredentialPayload) {
+    this.revertApiKey = process.env.REVERT_API_KEY || "";
+    this.revertApiUrl = process.env.REVERT_API_URL || "https://api.revert.dev/";
+    this.tenantId = String(credential.teamId ? credential.teamId : credential.userId); // Question: Is this a reasonable assumption to be made? Get confirmation on the exact field to be used here.
```

Comment:
**Question:** Is this a reasonable assumption to be made? 

Revert api expects a unique `tenantId` for each CRM/app. Since a CRM/app can be used by multiple users that are part of a team, can `teamId` be used here to uniquely identify a CRM?

Reply:
Looks right to me